### PR TITLE
[Woo POS] UI - Cart: Product rows

### LINKED ISSUE #12706: [Woo POS] UI: Product cards and product rows
Ref: TfaZ4LUkEwEGrxfnEFzvJj-fi-6_1657

- **Product cards:** should contain product name, plus button, dynamic quantity.
- **Product rows:** should contain product name, remove button.

Screenshot 2024-05-13 at 11 12 25

Tying them together through a common simple observable on a shared view model could be part of this PR as well, or tackled separately.

### LINKED ISSUE #12724: [Woo POS] UI: Dashboard CartProduct rows
Subtask from https://github.com/woocommerce/woocommerce-ios/issues/12703

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes #12724 
Closes #12706 

## Description
This PR updates the design for the product rows within the cart view, so they're closer to the initial wireframe. I've used the figma values of `120` for height, `56` for circled buttons (+, x, .. ), and `20`/`32` for padding.

Ref: TfaZ4LUkEwEGrxfnEFzvJj-fi-6_1657

Simulator Screen Recording - iPad Pro (12 9-inch) (6th generation) - 2024-05-16 at 10 16 34

## Testing instructions
* In POS mode, observe the changes on:
  * The `Cart` title aligned and using `.title`
  * The product rows 
  * The default-styled "Pay now" button is now a styled-"Checkout" button

Adding and removing products from the cart should behave normally.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommercePOS/WooCommercePOS/Presentation/CartView.swift
Comment (bozidarsevo) [inline WooCommercePOS/WooCommercePOS/Presentation/CartView.swift]: I suppose @ViewBuilder is not needed. Since there are not `if` statements etc.

It might be a personal preference, but I prefer having private vars for creating view components like this, so `private var checkoutButton: some View` instead of a `function.

File: WooCommercePOS/WooCommercePOS/Utils/Color+WooCommercePOS.swift
Comment (bozidarsevo) [inline WooCommercePOS/WooCommercePOS/Utils/Color+WooCommercePOS.swift]: Update the comment ;) I suppose it was a copy paste :D

File: WooCommercePOS/WooCommercePOS/Presentation/CartView.swift
Comment (iamgabrielma) [reply WooCommercePOS/WooCommercePOS/Presentation/CartView.swift]: > I suppose ViewBuilder is not needed.

You're right! Removed on https://github.com/woocommerce/woocommerce-ios/pull/12749/commits/8f221c4da995c459bb61b5fb5be4ff23adf96cba 

> It might be a personal preference, but I prefer having private vars for creating view components like this, so private var checkoutButton: some View instead of a `function.

Sounds good to me! Updated on: https://github.com/woocommerce/woocommerce-ios/pull/12749/commits/84e7f7f0999ab67d7ca82b461e9075323304fde9

File: WooCommercePOS/WooCommercePOS/Utils/Color+WooCommercePOS.swift
Comment (iamgabrielma) [reply WooCommercePOS/WooCommercePOS/Utils/Color+WooCommercePOS.swift]: Oops 😂 Fixed: https://github.com/woocommerce/woocommerce-ios/pull/12749/commits/84e7f7f0999ab67d7ca82b461e9075323304fde9

**Comment by iamgabrielma** [comment]
No need to review it for now @bozidarsevo , I'll wait until the framework has been removed to merge back trunk into this branch and fix any conflict that emerges. Most likely tomorrow 🙇

**Comment by iamgabrielma** [comment]
This is ready @bozidarsevo , the merge conflict was very straightforward and the only change was on renaming `POSPlusButtonStyle()`, so there are no changes from the items from previous review.

It currently targets 18.8, but feel free to update the milestone and merge it into 18.7 if you see this before the code freeze, otherwise we'll merge it next week! 👍

**Comment by bozidarsevo** [comment]
Not related to this change but `PointOfSaleDashboardViewModel` keeps getting recreated since `PointOfSaleEntryPointView is recreated. You can change it that `PointOfSaleEntryPointView` is initialized with the model that is a StateObject or something like that

Simulator Screen Recording - iPad Pro (11-inch) (4th generation) - 2024-05-20 at 10 11 56

**Review by bozidarsevo — APPROVED** [review]

**Comment by iamgabrielma** [comment]
Thanks for the review!

> Not related to this change but PointOfSaleDashboardViewModel keeps getting recreated since PointOfSaleEntryPointView is recreated. You can change it that PointOfSaleEntryPointView` is initialized with the model that is a StateObject or something like that

Yeah, we saw some problems last week with the views that seem related to this. I'll ping you in the thread, reference: p1715934814114829-slack-C025A8VV728

**Pull request merged**
